arbor/thread_private_spike_store.cpp

```diff
@@ -3,14 +3,15 @@
 #include <arbor/common_types.hpp>
 #include <arbor/spike.hpp>
 
+#include "threading/threading.hpp"
 #include "thread_private_spike_store.hpp"
 
 namespace arb {
 
 struct local_spike_store_type {
     threading::enumerable_thread_specific<std::vector<spike>> buffers_;
 
-    local_spike_store_type(const task_system_handle& ts): buffers_(ts) {};
+    local_spike_store_type(const task_system_handle& ts): buffers_(*ts) {};
```

Comment:
I changed the double buffer from taking a handle to just take a reference, because I don't see the point of passing around handles in the back end unless we want the receiver to store the thread pool.

With that in mind, I should change the constructor of `local_spike_store_type` to also just take a reference to a `task_system` if I am going to be consistent.

Do you think we should keep using the handles instead?

Reply:
I don't really mind, I was just interested in the reasoning :)

Reply:
This requires that the `enumerable_thread_local` class be moved into its own file, which is where it should have been all the time.